### PR TITLE
Skip inspecting pod network if unknown namespace

### PR DESCRIPTION
**Pull request by aaronlevy**
**What this PR does / why we need it**:

If we fail to determine the network namespace of a container we still try to inspect the state - even though there is no way for it to succeed. This leads to errors like:

> NetworkPlugin cni failed on the status hook for pod "X": Unexpected command output nsenter: cannot open : No such file or directory

Instead, if we cannot determine the network namespace, we should just exit with a (hopefully) more clear error message.

I left the wording as assuming a terminated pod, based on:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/helpers.go#L208-L211

ref: 
https://github.com/kubernetes-incubator/bootkube/issues/475
https://github.com/coreos/coreos-kubernetes/issues/856

### CODE REVIEW

File: pkg/kubelet/network/cni/cni.go
Comment (xiangpengzhao) [inline pkg/kubelet/network/cni/cni.go]: Finally we will get two same log info with different log levels, right? So I think we'd better modify the error info here.

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

File: pkg/kubelet/network/cni/cni.go
Comment (aaronlevy) [inline pkg/kubelet/network/cni/cni.go]: Yes, if you had `--v=4` you would see this message twice (once as info and ultimately once as a warning).

Maybe changing this to something like `Cannot find the network namespace, skipping pod network status` ?

**Comment by xiangpengzhao** [comment]
SGTM:)

**Comment by aaronlevy** [comment]
Updated the wording of the error messages.

**Comment by dcbw** [comment]
See also https://github.com/kubernetes/kubernetes/pull/44067 which is related but at a higher level.

There are two issues; this PR fixes one (don't inspect if we know we can't) while 44067 tries to fix the higher level issue, which is that kubelet shouldn't print an error if it knows that the pod doesn't actually exist (due to the PLEG race).

In any case, /lgtm.

**Comment by aaronlevy** [comment]
Probably doesn't need a release note.

Then also, after this merges I'd like to propose cherry-picking this into the v1.6 release branch

**Comment by aaronlevy** [comment]
Test failure seems to be unrelated. ping @bowei

**Comment by bowei** [comment]
@k8s-bot verify test this

**Comment by aaronlevy** [comment]
Bump

**Comment by aaronlevy** [comment]
@bowei @dnardo Sorry to keep bugging you guys - hoping to try and get this cherry-picked back into a v1.6 release -- it's tripping people up fairly consistently.

**Comment by bowei** [comment]
@freehan need approval

**Comment by aaronlevy** [comment]
Thanks! @bowei possible to add `cherrypick-candidate` label  (for v1.6.x)?

**Pull request merged**
